lib/cli/kit/system.rb

```diff
@@ -28,56 +32,54 @@ def sudo_reason(msg)
 
         # Execute a command in the user's environment
         # This is meant to be largely equivalent to backticks, only with the env passed in.
-        # Captures the results of the command without output to the console
-        #
-        # #### Parameters
-        # - `*a`: A splat of arguments evaluated as a command. (e.g. `'rm', folder` is equivalent to `rm #{folder}`)
-        # - `sudo`: If truthy, run this command with sudo. If String, pass to `sudo_reason`
-        # - `env`: process environment with which to execute this command
-        # - `**kwargs`: additional arguments to pass to Open3.capture2
-        #
-        # #### Returns
-        # - `output`: output (STDOUT) of the command execution
-        # - `status`: boolean success status of the command execution
-        #
-        # #### Usage
-        # `out, stat = CLI::Kit::System.capture2('ls', 'a_folder')`
-        #
-        def capture2(*a, sudo: false, env: ENV, **kwargs)
-          delegate_open3(*a, sudo: sudo, env: env, method: :capture2, **kwargs)
+        # Captures the results of the command without output to the console.
+        #
+        # Example:
+        #   out, stat = CLI::Kit::System.capture2('ls', 'a_folder')
+        sig do
+          params(
+            args: String, # splat of arguments ('rm dir' or 'rm', 'dir')
+            sudo: T.any(T::Boolean, String), # prompt for credentials and run as root?
+            env: T::Hash[String, String], # process environment with which to execute
+            kwargs: T.untyped, # additional args to pass to Open3.capture2
```

Comment:
Isn't that a `T.nilable[T::Array[String]]`?

Reply:
`T.nilable(T::Hash[Symbol, T.untyped])` but yeah, I'll add that.